sdk/data-generators/image-processing-generator/src/main.rs

```diff
@@ -29,82 +28,82 @@ fn main() -> Result<(), Box<dyn Error>> {
-               .long("seed")
-               .value_name("NUBMER")
-               .help("The seed for the random number generator.")
-               .takes_value(true)
-               .validator(is_u64)
-               .default_value("0"),
+        .about("Load a JPEG image and convert it to a PNG image of dimensions ([WIDTH], [HEIGHT])")
+        .arg(
+            Arg::with_name("file_prefix")
+                .short("f")
+                .long("file_prefix")
+                .value_name("STRING")
+                .help("The prefix for the output file")
+                .takes_value(true)
+                .required(true),
+        )
+        .arg(
+            Arg::with_name("image_path")
+                .short("i")
+                .long("image_path")
+                .value_name("STRING")
+                .help("The path to the JPEG image to load")
+                .takes_value(true)
+                .required(true),
+        )
+        .arg(
+            Arg::with_name("width")
+                .short("w")
+                .long("width")
+                .value_name("NUMBER")
+                .help("The width of the image to generate")
+                .takes_value(true)
+                .validator(is_u64)
+                .default_value("10"),
+        )
+        .arg(
+            Arg::with_name("height")
+                .short("h")
+                .long("height")
+                .value_name("NUMBER")
+                .help("The height of the image to generate")
+                .takes_value(true)
+                .validator(is_u64)
+                .default_value("10"),
         )
         .get_matches();
 
     let file_prefix = matches
         .value_of("file_prefix")
         .ok_or("Failed to read the file prefix.")?;
+    let image_path = matches
+        .value_of("image_path")
+        .ok_or("Failed to read the image path.")?;
     let image_width = matches
         .value_of("width")
-        .ok_or("Failed to read the width")?;
+        .ok_or("Failed to read the width")?
+        .parse::<u64>()
+        .map_err(|_| "Failed to parse the width.")? as u32;
     let image_height = matches
         .value_of("height")
-        .ok_or("Failed to read the height")?;
+        .ok_or("Failed to read the height")?
+        .parse::<u64>()
+        .map_err(|_| "Failed to parse the height.")? as u32;
 
-    // Download random image
-    let mut downloader = Downloader::builder()
-        .download_folder(std::path::Path::new("./"))
-        .parallel_requests(1)
-        .build()
-        .expect("Failed to configure the downloader");
-    let file_path = format!("{}.dat", file_prefix);
-    let dl =
-        Download::new(format!("https://picsum.photos/{}/{}", image_width, image_height).as_str())
-            .file_name(Path::new(&file_path));
-    let _result = downloader
-        .download(&[dl])
-        .map_err(|e| format!("Failed to download image: {}", e));
+    let output_filename = format!("{}.dat", file_prefix);
```

Comment:
Why `.dat` instead of `.png`?

Reply:
That's the pattern followed by every other generator, that being said they generate numbers, not images

Reply:
Yeah, I think in this case it makes more sense to use the established `png` file extension, as all other examples are creating *sui generis* data formats from `serde` serialisations of data structures.